File: Тренировка 1.0/03. Множества/F. Инопланетный геном/Инопланетный геном.cs
using System;
using System.Collections.Generic;

string gen1 = Console.ReadLine();
string gen2 = Console.ReadLine();

HashSet<string> set = new HashSet<string>();
for(int i = 0; i < gen2.Length - 1; i++) set.Add(gen2[i].ToString() + gen2[i + 1]);

int count = 0;
for(int i = 0; i < gen1.Length - 1; i++)
    if (set.Contains(gen1[i].ToString() + gen1[i + 1])) count++;

Console.WriteLine(count);

File: Тренировка 1.0/03. Множества/F. Инопланетный геном/Инопланетный геном.cpp
#include <iostream>
#include <string>
#include <unordered_set>

int main()
{
    std::string gen1, gen2;
    std::cin >> gen1 >> gen2;

    std::string pair;
    std::unordered_set<std::string> set;
    for (int i = 0; i < gen2.size() - 1; ++i)
    {
        pair = gen2[i];
        pair += gen2[i + 1];
        set.emplace(pair);
    }

    int count = 0;
    for (int i = 0; i < gen1.size() - 1; ++i)
    {
        pair = gen1[i];
        pair += gen1[i + 1];
        if (set.count(pair)) count++;
    }

    std::cout << count;
}
